runtime/common/BaseRemoteRESTQPU.h

```diff
@@ -382,12 +382,13 @@ class BaseRemoteRESTQPU : public cudaq::QPU {
     moduleOp.push_back(func.clone());
     moduleOp->setAttrs(m_module->getAttrDictionary());
 
-    // Add any global symbols associated with custom operations (available after
-    // `lift-array-value` and `get-concrete-matrix` passes)
     for (auto &op : m_module.getOps()) {
-      if (auto ccGlobalOp = dyn_cast<cudaq::cc::GlobalOp>(op)) {
-        moduleOp.push_back(ccGlobalOp.clone());
-      }
+      // Add any global symbols, including global constant arrays.
+      // Global constant arrays can be created during compilation,
+      // `lift-array-value`, `quake-synthesizer`, and `get-concrete-matrix`
+      // passes.
+      if (auto globalOp = dyn_cast<cudaq::cc::GlobalOp>(op))
+        moduleOp.push_back(globalOp.clone());
```

Comment:
Not this PR, but a random comment. I'm not sure why we're trying to cherry-pick pieces out of the module here. It seems to create problems and if we just copied the entire module, what would be the downside? (Asking for a friend.)

Reply:
Pradnya had the same changes too - the code was already picking a function for a new module, but did not copy globals, which we need now. Note for future - we might want to only add globals used in the function. Could this be because we send one function at a time somewhere? 

@bmhowe23 do you know why we are not just using the whole module?

Reply:
> @bmhowe23 do you know why we are not just using the whole module?

I don't know for sure, but it might be that the server doesn't like receiving multiple entry-point functions. That's pure speculation, though. I would need to do more investigation to be sure.